### PR TITLE
Allow policy update of SGI, in case if the manager is supporting multiple policies and only one policy is binded

### PR DESCRIPTION
**Pull request by larkinscarvalho**

### CODE REVIEW

**Comment by arvindn05** [comment]
please add referencce to bug #551 
also update the bug with scenrios where the box is enabled/disabled

**Comment by larkinscarvalho** [comment]
Fixes: https://github.com/opensecuritycontroller/osc-core/issues/551

If we have one policy bounded, we should be able to update the policies and the checkbox is enabled.

However, if we have a SGI with multiple policies bounded, we cannot update the policies and the checkbox is disabled.

**Comment by arvindn05** [comment]
Just to confirm, unbinding the policy/SGI is allowed in all cases right?

File: osc-ui/src/main/java/org/osc/core/broker/view/vc/securitygroup/BindSecurityGroupWindow.java [hunk 1]
```diff
@@ -418,9 +418,10 @@ private void serviceTableClicked(long itemId) {
 		}
 	}
 
-	private boolean isMultiplePoliciesSupported(Long vsId) {
+	private boolean isBindingWithMultiplePolicies(Long vsId) {
```
Comment (arvindn05) [inline osc-ui/src/main/java/org/osc/core/broker/view/vc/securitygroup/BindSecurityGroupWindow.java]: isBindedWithMultiplePolicies

File: osc-ui/src/main/java/org/osc/core/broker/view/vc/securitygroup/BindSecurityGroupWindow.java [hunk 2]
```diff
@@ -418,9 +413,10 @@ private void serviceTableClicked(long itemId) {
 		}
 	}
 
-	private boolean isMultiplePoliciesSupported(Long vsId) {
+	private boolean isBindedWithMultiplePolicies(Long vsId) {
 		for (VirtualSystemPolicyBindingDto binding : this.allBindings) {
-			if (binding.getVirtualSystemId().equals(vsId) && binding.isMultiplePoliciesSupported()) {
+			if (binding.getVirtualSystemId().equals(vsId) && binding.isMultiplePoliciesSupported() && binding.isBinded()
```
Comment (arvindn05) [inline osc-ui/src/main/java/org/osc/core/broker/view/vc/securitygroup/BindSecurityGroupWindow.java]: can we extract the condition below into another method and use it in line 152 as well?

`binding.isMultiplePoliciesSupported() && binding.isBinded() && binding.getPolicyIds().size() > 1`

we can name the overloaded method the same and add javadoc to both methods.
`isBindedWithMultiplePolicies(VirtualSystemPolicyBindingDto binding)`

**Comment by larkinscarvalho** [comment]
I tried it already.
But, to find and match the virtual system, becomes messy.

**Comment by arvindn05** [comment]
@larkinscarvalho not sure i understand. you will have 2 methods called isBindedWithMultiplePolicies() should be straightforward..

your condition will become

`binding.getVirtualSystemId().equals(vsId) && isBindedWithMultiplePolicies(binding)`

File: osc-ui/src/main/java/org/osc/core/broker/view/vc/securitygroup/BindSecurityGroupWindow.java [hunk 3]
```diff
@@ -418,9 +413,10 @@ private void serviceTableClicked(long itemId) {
 		}
 	}
 
-	private boolean isMultiplePoliciesSupported(Long vsId) {
+	private boolean isBindedWithMultiplePolicies(Long vsId) {
 		for (VirtualSystemPolicyBindingDto binding : this.allBindings) {
```
Comment (larkinscarvalho) [inline osc-ui/src/main/java/org/osc/core/broker/view/vc/securitygroup/BindSecurityGroupWindow.java]: @arvindn05 we iterate on **this.allBindings** to find the right binding. If we want to introduce change, we will have to update serviceTableClicked to find the right bind and provide to isBindedWithMultiplePolicies

Comment (larkinscarvalho) [inline osc-ui/src/main/java/org/osc/core/broker/view/vc/securitygroup/BindSecurityGroupWindow.java]: @arvindn05 we iterate on this.allBindings to find the right binding. If we want to introduce change, we will have to update serviceTableClicked to find the right bind and provide to isBindedWithMultiplePolicies

Comment (arvindn05) [reply osc-ui/src/main/java/org/osc/core/broker/view/vc/securitygroup/BindSecurityGroupWindow.java]: i dont see the issue...you will still iterate in the isBindedWithMultiplePolicies(Long vsId) method...so no changes should be needed.

**Comment by larkinscarvalho** [comment]
@arvindn05 ready for CR. Thanks.

**Pull request merged**